std/range/package.d

```diff
@@ -7125,9 +7131,57 @@ private:
     size_t _chunkSize;
 }
 
+/// ditto
+struct Chunks(Source)
+if (isInputRange!Source && !isForwardRange!Source)
+{
+    static struct Chunk
+    {
+        private Chunks* impl;
+
+        @property bool empty() { return impl.curSizeLeft == 0 || impl.r.empty; }
```

Comment:
Could be `const` if `impl.r.empty` is const. This is typically checked with sth. like:

```
static if (is(typeof((cast(const typeof(impl.r))impl.r).empty))) 
```

but it's not very clean and thus not done at many places in Phobos ...